src/Microsoft.AspNet.Routing/RouteValueDictionary.cs

```diff
@@ -43,46 +39,63 @@ public RouteValueDictionary()
         /// Only public instance non-index properties are considered.
         /// </remarks>
         public RouteValueDictionary(object values)
-            : this()
         {
-            if (values != null)
+            var otherDictionary = values as RouteValueDictionary;
+            if (otherDictionary != null)
             {
-                var keyValuePairCollection = values as IEnumerable<KeyValuePair<string, object>>;
-                if (keyValuePairCollection != null)
+                if (otherDictionary.InnerDictionary != null)
                 {
-                    foreach (var kvp in keyValuePairCollection)
+                    InnerDictionary = new Dictionary<string, object>(
```

Comment:
Could you add a note to this saying this was done to prevent boxing the Dictionary enumerator?